bench/protobuf/optimized/padding_optimizer.ll

```diff
@@ -2017,14 +2016,16 @@ common.ret21:                                     ; preds = %if.end, %if.then
   ret void
 
 if.end:                                           ; preds = %entry
-  %div1920 = lshr i64 %sub.ptr.div.i, 1
-  %add.ptr.i = getelementptr inbounds nuw %"class.google::protobuf::compiler::cpp::(anonymous namespace)::FieldGroup", ptr %__first.coerce, i64 %div1920
+  %div1920 = lshr exact i64 %sub.ptr.sub.i, 1
+  %add.ptr.i.idx = and i64 %div1920, 4611686018427387872
```

Comment:
A bunch of changes like this that look regression-y. Here is what is going on: https://llvm.godbolt.org/z/dhfPT1f3b

Worth noting that we get an `and` in the final codegen anyway so that in itself is fine. The fact that demanded bits mangles the constant based on the knowledge that the top bits are zero is unfortunate though, because it makes the immediate encoding less efficient.